File: Greedy/435. Non-overlapping Intervals/435. Non-overlapping Intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        if (intervals == null || intervals.length == 0 || intervals[0].length == 0) return 0;
        Arrays.sort(intervals, (a, b) -> (a[0] == b[0] ? a[1] - b[1] : a[0] - b[0]) );
        
        int rets = 0;
        int[] curStand = Arrays.copyOf(intervals[0], 2);
        for (int i = 1; i < intervals.length; i++) {
            // if  overlap
            if (intervals[i][0] < curStand[1]) {
                rets += 1;   
            } else {
                curStand[0] = intervals[i][0];
                curStand[1] = intervals[i][1];
            }
            if (intervals[i][1] < curStand[1]) {
                curStand[0] = intervals[i][0];
                curStand[1] = intervals[i][1];
            }
        }
        return rets;
    }
}

File: Greedy/435. Non-overlapping Intervals/435. Non-overlapping Intervals.cpp
class Solution {
    static bool cmp(vector<int>& a, vector<int>& b) {
        return a[1] < b[1];
    }
    
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), cmp);
        
        int n = intervals.size();
        int i = 0;
        int count = 0;
        while (i < n) {
            count++;
            int j = i+1;
            while (j < n && intervals[j][0] < intervals[i][1]) {
                j++;
            }
            i = j;
        }
        return n - count;
    }
};
